base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -452,7 +447,22 @@ where B: BlockchainBackend + 'static
             );
             return Ok(());
         }
-
+        {
+            // we have this in mutex to ensure we dont have any funnies with rwlock with multiples requests first
+            // passing the check then all wanting to reconcile. We should only have 1 request passing
```

Comment:
We use a mutex because we have multiple threads interrogating the hashset at the same time. The hashset is the important part. We allow _different_ blocks to be reconciled concurrently. 
```suggestion
            //  This handler is called in its own task for each new block message. The same block message may
            // be received from multiple peers as a normal part of block propagation. We use a HashSet to prevent 
            // the same block from being reconciled at the same time before it has a chance to be added to the blockchain.
```